monai/apps/deepgrow/dataset.py

```diff
@@ -0,0 +1,271 @@
+            data_list.append(
+                {
+                    "image": image_file.replace(dataset_dir + "/", "") if relative_path else image_file,
+                    "label": label_file.replace(dataset_dir + "/", "") if relative_path else label_file,
+                    "region": int(idx),
+                }
+            )
+
+    print(
+        "{} => Image: {} => {}; Label: {} => {}; Unique Labels: {}".format(
+            vol_idx,
+            vol_image.shape,
+            image_count,
+            vol_label.shape if vol_label is not None else None,
+            label_count,
+            unique_labels_count,
+        )
+    )
+    return data_list
+
+
+def _save_data_3d(vol_idx, data, keys, dataset_dir, relative_path):
+    vol_image = data[keys[0]]
+    vol_label = data.get(keys[1])
+    data_list = []
+
+    if len(vol_image.shape) == 4:
+        logging.info("4D-Image, pick only first series; Image: {}; Label: {}".format(vol_image.shape, vol_label.shape))
+        vol_image = vol_image[0]
+        vol_image = np.moveaxis(vol_image, -1, 0)
+
+    image_count = 0
+    label_count = 0
+    unique_labels_count = 0
+
+    image_file_prefix = "vol_idx_{:0>4d}".format(vol_idx)
+    image_file = os.path.join(dataset_dir, "images", image_file_prefix)
+    image_file += ".npy"
+
+    os.makedirs(os.path.join(dataset_dir, "images"), exist_ok=True)
+    np.save(image_file, vol_image)
+    image_count += 1
+
+    # Test Data
+    if vol_label is None:
+        data_list.append(
+            {
+                "image": image_file.replace(dataset_dir + "/", "") if relative_path else image_file,
+            }
+        )
+    else:
+        # For all Labels
+        unique_labels = np.unique(vol_label.flatten())
+        unique_labels = unique_labels[unique_labels != 0]
+        unique_labels_count = max(unique_labels_count, len(unique_labels))
+
+        for idx in unique_labels:
+            label_file_prefix = "{}_region_{:0>2d}".format(image_file_prefix, int(idx))
+            label_file = os.path.join(dataset_dir, "labels", label_file_prefix)
+            label_file += ".npy"
+
+            curr_label = (vol_label == idx).astype(np.float32)
+            os.makedirs(os.path.join(dataset_dir, "labels"), exist_ok=True)
+            np.save(label_file, curr_label)
+
+            label_count += 1
+            data_list.append(
+                {
+                    "image": image_file.replace(dataset_dir + "/", "") if relative_path else image_file,
+                    "label": label_file.replace(dataset_dir + "/", "") if relative_path else label_file,
+                    "region": int(idx),
+                }
+            )
+
+    print(
+        "{} => Image: {} => {}; Label: {} => {}; Unique Labels: {}".format(
+            vol_idx,
+            vol_image.shape,
+            image_count,
+            vol_label.shape if vol_label is not None else None,
+            label_count,
+            unique_labels_count,
+        )
+    )
```

Comment:
this print need to be either more descriptive or we remove them or we put it as debug msgs?

Reply:
fixed